### ISSUE TITLE
'Move task down' does not maintain focus when using Alt+ArrowDown as a custom shortcut

### ISSUE THREAD

**Issue by mart3323** [johannesjo/super-productivity#2650]
### Your Environment

- **Version used**: Reproduced on webapp at https://app.super-productivity.com/, version 7.13.2 Rev 311c464eaf9ceb628684cff24b2fe038ff7021ef
- **Operating System and version**: Windows 10 Pro 19045.3086
- **Browser Name and version**: Google Chrome 114.0.5735.134 (Official Build) (64-bit)

### Expected Behavior
After having changed the shortcuts for moving tasks to be <kbd>alt</kbd>+<kbd>↑ ↓</kbd>, moving the currently selected task down should keep that task selected

### Current Behavior
Moving the task down will *also* move the selection to the first subtask / the next task

### Steps to Reproduce (for bugs)

1. Open super productivity (reproduced on webapp but same issue happens if installed as an app)
2. Change the shortcuts for moving tasks  
 <img src="https://github.com/johannesjo/super-productivity/assets/8430090/27b2010c-27d0-401a-9727-4def2db5a349" height="100px"/>  

3. Create three tasks as follows, and select the topmost task
<img src="https://github.com/johannesjo/super-productivity/assets/8430090/c1e23cf9-3056-4f42-b3ce-4f28c6991652" height="150px"/>  

4. Hit the shortcut to move the task down (<kbd>alt</kbd>+<kbd> ↓</kbd>)  
❌ (expected task 3 to remain selected)
<img src="https://github.com/johannesjo/super-productivity/assets/8430090/01cc6761-c635-404a-83fb-9d0037fc9361" height="150px"/>  

5. Hit the shortcut to move the task down (<kbd>alt</kbd>+<kbd> ↓</kbd>)   
❌ (expected task 2 to remain selected)
<img src="https://github.com/johannesjo/super-productivity/assets/8430090/dadb7a17-ae57-4104-935f-32c59a9e89c2" height="150px"/>  

6. Hit the shortcut to move the task down (<kbd>alt</kbd>+<kbd> ↓</kbd>)   
❌ (expected subtask 1 to remain selected)
<img src="https://github.com/johannesjo/super-productivity/assets/8430090/22ddd4fc-7915-4698-a3e2-59711892f276" height="150px"/>  

### Can you reproduce this reliably?
Yes, but only with my custom shortcut, the default ctrl-shift one works fine

### Console Output
<details><summary>Probably nothing relevant</summary>

```
action-logger.reducer.ts:11 [WorkContextMeta] Move Task Down in Today Object
action-logger.reducer.ts:11 [Persistence] Save to DB Object
action-logger.reducer.ts:11 [WorkContextMeta] Move Task Up in Today Object
action-logger.reducer.ts:11 [Persistence] Save to DB Object
action-logger.reducer.ts:11 [WorkContextMeta] Move Task Down in Today Object
action-logger.reducer.ts:11 [Persistence] Save to DB Object
action-logger.reducer.ts:11 [WorkContextMeta] Move Task Down in Today ObjectdoneTaskIds: (3) ['c8SJrdZfJjOCl29JrdlIc', 'yu3ujPtxVRaJeFR7kNbGd', 'B39re2PDXRDyP1ySPSVs9']taskId: "c8SJrdZfJjOCl29JrdlIc"type: "[WorkContextMeta] Move Task Down in Today"workContextId: "TODAY"workContextType: "TAG"[[Prototype]]: Object
action-logger.reducer.ts:11 [Persistence] Save to DB Objectdata: {ids: Array(1), entities: {…}, __modelVersion: 1}dbKey: "tag"type: "[Persistence] Save to DB"[[Prototype]]: Object
action-logger.reducer.ts:11 [Task] Move down Objectid: "bX-lQJ6zgeDMWYZspDV4y"parentId: "c8SJrdZfJjOCl29JrdlIc"type: "[Task] Move down"[[Prototype]]: Object
action-logger.reducer.ts:11 [Persistence] Save to DB Objectdata: {ids: Array(5), entities: {…}, currentTaskId: null, selectedTaskId: null, taskAdditionalInfoTargetPanel: 'Default', …}dbKey: "task"type: "[Persistence] Save to DB"[[Prototype]]: Object
```

</details>

**Comment by mart3323** [comment]
I found that there's a workaround implemented in c152c96c6fa2d8df602ae0f7d21b106b075c2d2a but only for the 'move up' behaviour. Copying that to move down as well seems to fix the issue for me
<details><summary>code</summary>

```diff
    // moving items
    // move task up
    if (checkKeyCombo(ev, keys.moveTaskUp)) {
      this._taskService.moveUp(this.task.id, this.task.parentId, this.isBacklog);
      ev.stopPropagation();
      ev.preventDefault();
      // timeout required to let changes take place @TODO hacky
-     setTimeout(this.focusSelf.bind(this), 50);
+     setTimeout(this.focusSelf.bind(this), 0);
    }
+  // move task down
    if (checkKeyCombo(ev, keys.moveTaskDown)) {
      this._taskService.moveDown(this.task.id, this.task.parentId, this.isBacklog);
      ev.stopPropagation();
      ev.preventDefault();
+     // timeout required to let changes take place @TODO hacky
+     setTimeout(this.focusSelf.bind(this), 0);
    }
  }
```

</details>

Alternatively, i noticed that the focus actions are hardcoded to skip if shift or ctrl is being held. If i change that condition to be based on the moveTask shortcuts instead then this *also* fixes the issue for me

<details><summary>code</summary>

```diff
+   const isMoveTaskCombo = checkKeyCombo(ev, keys.moveTaskUp) || checkKeyCombo(ev, keys.moveTaskDown)
-   const isShiftOrCtrlPressed = ev.shiftKey || ev.ctrlKey;


    // move focus up
    if (
-     (!isShiftOrCtrlPressed && ev.key === 'ArrowUp') ||
+     (!isMoveTaskCombo && ev.key === 'ArrowUp') ||
      checkKeyCombo(ev, keys.selectPreviousTask)
    ) {
      ev.preventDefault();
      this.focusPrevious();
    }
    // move focus down
    if (
-     (!isShiftOrCtrlPressed && ev.key === 'ArrowDown') ||
+     (!isMoveTaskCombo && ev.key === 'ArrowDown') ||
      checkKeyCombo(ev, keys.selectNextTask)
    ) {
      ev.preventDefault();
      this.focusNext();
    }
```

</details>

**Comment by johannesjo** [comment]
Thanks for your input! Ill investigate this further!

**Comment by mart3323** [comment]
According to changelogs, resolved by 13c03b1ddf6a1e00f78d401f009d6544b466dd15 in [7.14.0](https://github.com/johannesjo/super-productivity/compare/v7.13.1...v7.14.0).

Although i'm not sure how to test - i still see 7.13.2 in both the desktop app as well as the web app. Is that expected?

**Comment by mart3323** [comment]
Tested using the .exe file from releases, working as expected! 🎉
